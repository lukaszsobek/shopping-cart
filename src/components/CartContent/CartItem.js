import React from "react";
import Button from "../shared/Button";
import { currencyRound } from "../../utils";


const CartItem = (props) => {
    const {
        comments,
        deleteItem,
        id,
        name,
        net_price,
        number,
        openEditItemModal,
        tax
    } = props;

    const tax_string = `${tax}%`;
    const gross_price = net_price + ( net_price * tax / 100);

    return (
        <div className="cart-item">
            <div className="item-number">{ number }</div>
            <div className="cart-item-details">
                <div className="item-name">{ name }</div>
                <div className="item-comments">{ comments }</div>
                <div className="item-net-price">{ net_price }&euro;</div>
                <div className="item-tax">{ tax_string }</div>
                <div className="item-gross-price">{ currencyRound(gross_price) }&euro;</div> 
            </div>
            <div className="action-buttons">

                <Button
                    action={ () => openEditItemModal(id) }
                    className="edit-item-button"
                    label="..."
                />
                <Button
                    action={ () => deleteItem(id) }
                    className="delete-item-button"
                    label="x"
                />
            </div>
        </div>
    );
};

export default CartItem;