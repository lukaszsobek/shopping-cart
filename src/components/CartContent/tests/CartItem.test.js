import CartItem from "../CartItem";

describe("Renders",() => {
    it("cart item", () => {

        const props = {
            comments: "test",
            deleteItem: () => null,
            id: 0,
            name: "some item",
            net_price: 10.28,
            number: 1,
            openEditItemModal: () => null,
            tax: 7
        }

        expect(CartItem(props)).toMatchSnapshot();
    })
})