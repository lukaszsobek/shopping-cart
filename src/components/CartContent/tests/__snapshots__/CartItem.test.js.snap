// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders cart item 1`] = `
<div
  className="cart-item"
>
  <div
    className="item-number"
  >
    1
  </div>
  <div
    className="item-name"
  >
    some item
  </div>
  <div
    className="item-comments"
  >
    test
  </div>
  <div
    className="item-net-price"
  >
    10.28
    €
  </div>
  <div
    className="item-tax"
  >
    7%
  </div>
  <div
    className="item-gross-price"
  >
    11
    €
  </div>
  <div
    className="action-buttons"
  >
    <Button
      action={[Function]}
      className="edit-item-button"
      label="✎"
    />
    <Button
      action={[Function]}
      className="delete-item-button"
      label="x"
    />
  </div>
</div>
`;
